File: exercises/exercise-1-testing-java/src/main/java/io/miragon/example/application/port/inbound/bike/StartBikeSubscriptionUseCase.java
package io.miragon.example.application.port.inbound.bike;

import io.miragon.example.domain.Email;
import io.miragon.example.domain.Name;
import io.miragon.example.domain.bike.BikeId;
import io.miragon.example.domain.bike.BikeSubscriptionId;

public interface StartBikeSubscriptionUseCase {
    record Command(
        BikeId bikeId,
        Email email,
        Name name
    ) {}

    BikeSubscriptionId start(Command command);
}

File: exercises/exercise-1-testing-kotlin/src/main/kotlin/io/miragon/example/application/port/inbound/bike/StartBikeSubscriptionUseCase.kt
package io.miragon.example.application.port.inbound.bike

import io.miragon.example.domain.Email
import io.miragon.example.domain.Name
import io.miragon.example.domain.bike.BikeId
import io.miragon.example.domain.bike.BikeSubscriptionId

interface StartBikeSubscriptionUseCase {
    data class Command(
        val bikeId: BikeId,
        val email: Email,
        val name: Name
    )

    fun start(command: Command): BikeSubscriptionId
}
